### PR TITLE
Support Android 10+

### PR DESCRIPTION
**Pull request by N-X-T**
- This is the version that supports android 10+. Hope you will integrate android 10, 11 support in the lastest version.
- There is one thing to note: In the RefBase.h file, Android 10, 11 does not have the `incWeakRequireWeak()` method, I replaced it with `incWeak()` and it works with Android 10, 11 and 12+. I'm not sure what's the difference between the above 2 methods, there's little documentation about them.

### CODE REVIEW

**Comment by YT-Advanced** [comment]
image

Can TrickyStore being an open source project again after this PRs 👀

File: service/src/main/java/io/github/a13e300/tricky_store/KeystoreInterceptor.kt [hunk 1]
```diff
@@ -1,27 +1,49 @@
...
+    private val getTransaction =
+        getTransactCode(IKeystoreService.Stub::class.java, "get")
+    private val generateKeyTransaction =
+        getTransactCode(IKeystoreService.Stub::class.java, "generateKey")
+    private val getKeyCharacteristicsTransaction =
+        getTransactCode(IKeystoreService.Stub::class.java, "getKeyCharacteristics")
+    private val exportKeyTransaction =
+        getTransactCode(IKeystoreService.Stub::class.java, "exportKey")
+    private val attestKeyTransaction =
+        getTransactCode(IKeystoreService.Stub::class.java, "attestKey")
```
Comment (5ec1cff) [inline service/src/main/java/io/github/a13e300/tricky_store/KeystoreInterceptor.kt:37]: Why do we need to intercept these transactions?

File: service/src/main/java/io/github/a13e300/tricky_store/KeystoreInterceptor.kt [hunk 2]
```diff
@@ -65,23 +267,27 @@ object KeystoreInterceptor : BinderInterceptor() {
...
         val p = Parcel.obtain()
         Logger.d("intercept post $target uid=$callingUid pid=$callingPid dataSz=${data.dataSize()} replySz=${reply.dataSize()}")
         try {
-            val response = reply.readTypedObject(KeyEntryResponse.CREATOR)
-            val chain = Utils.getCertificateChain(response)
-            if (chain != null) {
-                val newChain = CertHack.hackCertificateChain(chain)
-                Utils.putCertificateChain(response, newChain)
-                Logger.i("hacked cert of uid=$callingUid")
+            data.setDataPosition(104)
```
Comment (5ec1cff) [inline service/src/main/java/io/github/a13e300/tricky_store/KeystoreInterceptor.kt]: What does this (L275) mean?

Comment (N-X-T) [reply service/src/main/java/io/github/a13e300/tricky_store/KeystoreInterceptor.kt]: L275 can change to `data.enforceInterface(DESCRIPTOR)`.
after `data.enforceInterface(DESCRIPTOR)`, dataPosition jump to 104

File: service/src/main/java/io/github/a13e300/tricky_store/KeystoreInterceptor.kt (hunk 1 above)
Comment (N-X-T) [reply service/src/main/java/io/github/a13e300/tricky_store/KeystoreInterceptor.kt:37]: Keypair generation in android 10,11 at  `AndroidKeyStoreKeyPairGeneratorSpi.generateKeyPair` same Android 12+ but there is a difference in the source code.
- It's start here: [generateKeystoreKeyPair](https://cs.android.com/android/platform/superproject/+/android-11.0.0_r21:frameworks/base/keystore/java/android/security/keystore/AndroidKeyStoreKeyPairGeneratorSpi.java;drc=b3a13e1e2ceb8ed7d607a5fc146a799438bfa52e;l=471): call IKeystoreService.generateKey(), return errorCode - generateKeyTransaction
- then: [loadKeystoreKeyPair](https://cs.android.com/android/platform/superproject/+/android-11.0.0_r21:frameworks/base/keystore/java/android/security/keystore/AndroidKeyStoreKeyPairGeneratorSpi.java;drc=b3a13e1e2ceb8ed7d607a5fc146a799438bfa52e;l=473) -> [AndroidKeyStoreProvider.loadAndroidKeyStoreKeyPairFromKeystore()](https://cs.android.com/android/platform/superproject/+/android-11.0.0_r21:frameworks/base/keystore/java/android/security/keystore/AndroidKeyStoreProvider.java;drc=7651f0a4c059a98f32b0ba30cd64500bf135385f;l=326) ->  in param: [getKeyCharacteristics()](https://cs.android.com/android/platform/superproject/+/android-11.0.0_r21:frameworks/base/keystore/java/android/security/keystore/AndroidKeyStoreProvider.java;drc=7651f0a4c059a98f32b0ba30cd64500bf135385f;l=229) -> IKeystoreService..getKeyCharacteristics() - return errorCode **getKeyCharacteristicsTransactions**; after have param: [loadAndroidKeyStorePublicKeyFromKeystore](https://cs.android.com/android/platform/superproject/+/android-11.0.0_r21:frameworks/base/keystore/java/android/security/keystore/AndroidKeyStoreProvider.java;drc=7651f0a4c059a98f32b0ba30cd64500bf135385f;l=250) -> IKeystoreService.exportKey() - return exportResult with resultCode
- last: [storeCertificateChain()](https://cs.android.com/android/platform/superproject/+/android-11.0.0_r21:frameworks/base/keystore/java/android/security/keystore/AndroidKeyStoreKeyPairGeneratorSpi.java;drc=b3a13e1e2ceb8ed7d607a5fc146a799438bfa52e;l=475) -> in param call [createCertificateChain()](https://cs.android.com/android/platform/superproject/+/android-11.0.0_r21:frameworks/base/keystore/java/android/security/keystore/AndroidKeyStoreKeyPairGeneratorSpi.java;drc=b3a13e1e2ceb8ed7d607a5fc146a799438bfa52e;l=492) -> [getAttestationChain()](https://cs.android.com/android/platform/superproject/+/android-11.0.0_r21:frameworks/base/keystore/java/android/security/keystore/AndroidKeyStoreKeyPairGeneratorSpi.java;drc=b3a13e1e2ceb8ed7d607a5fc146a799438bfa52e;l=604) -> IKeystoreService.attestKey() - return errorCode

All these transactions is return errorCode, if errorCode != 1, it's throw TEE broken

File: service/src/main/java/io/github/a13e300/tricky_store/KeystoreInterceptor.kt (hunk 2 above)
Comment (N-X-T) [reply service/src/main/java/io/github/a13e300/tricky_store/KeystoreInterceptor.kt]: I have pushed 1 commit clean code at L275 and remove useless code

**Comment by aviraxp** [comment]
Thx for your contribution.

**Comment by NCLnclNCL** [comment]
> * This is the version that supports android 10+. Hope you will integrate android 10, 11 support in the lastest version.
> * There is one thing to note: In the RefBase.h file, Android 10, 11 does not have the `incWeakRequireWeak()` method, I replaced it with `incWeak()` and it works with Android 10, 11 and 12+. I'm not sure what's the difference between the above 2 methods, there's little documentation about them.

Bro try add support android 8, 9

**Pull request closed without merging**